src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/MergeService.kt

```diff
@@ -76,25 +104,34 @@ class MergeService(
         EventKeys.TO_SOURCE_SYSTEM to mergeEvent.mergedRecord.sourceSystemType.name,
       ),
     )
-    updateTargetRecord(mergeEvent, targetPersonEntity)
+    mergeRecord(mergeEvent, null, targetPersonEntity) { _, targetPerson ->
+      updateTargetRecord(mergeEvent, targetPerson)
+    }
   }
 
-  private fun mergeRecord(mergeEvent: MergeEvent, sourcePersonEntity: PersonEntity?, targetPersonEntity: PersonEntity) {
-    when {
-      sourcePersonEntity == null -> handleSourceRecordNotFound(mergeEvent, targetPersonEntity)
-      else -> updateAndLinkRecords(mergeEvent, sourcePersonEntity, targetPersonEntity)
-    }
+  private fun mergeRecord(mergeEvent: MergeEvent, sourcePersonEntity: PersonEntity?, targetPersonEntity: PersonEntity, mergeAction: (sourcePersonEntity: PersonEntity?, targetPersonEntity: PersonEntity) -> Unit) {
+    val initialSourceUuid = sourcePersonEntity?.personKey?.let { it.personId.toString() }
+    val initialTargetUuid = targetPersonEntity.personKey?.let { it.personId.toString() }
+    mergeAction(sourcePersonEntity, targetPersonEntity)
     telemetryService.trackEvent(
       CPR_RECORD_MERGED,
       mapOf(
-        EventKeys.TO_UUID to targetPersonEntity.personKey?.let { it.personId.toString() },
-        EventKeys.FROM_UUID to sourcePersonEntity?.personKey?.let { it.personId.toString() },
-        EventKeys.TO_SOURCE_SYSTEM to targetPersonEntity.sourceSystem.name,
-        EventKeys.FROM_SOURCE_SYSTEM to sourcePersonEntity?.sourceSystem?.name,
+        EventKeys.TO_UUID to initialTargetUuid,
+        EventKeys.FROM_UUID to initialSourceUuid,
+        mergeEvent.sourceSystemId.first to mergeEvent.sourceSystemId.second,
+        mergeEvent.targetSystemId.first to mergeEvent.targetSystemId.second,
+        EventKeys.FROM_SOURCE_SYSTEM to mergeEvent.mergedRecord.sourceSystemType.name,
```

Comment:
Is this correct? Looks like FROM_SOURCE_SYSTEM will always be equal to  TO_SOURCE_SYSTEM, in which case why log both?

Reply:
Yeah fair point! Ill update in subsequent PR